filebeat/harvester/reader/json.go

```diff
@@ -125,3 +127,10 @@ func MergeJSONFields(data common.MapStr, jsonFields common.MapStr, text *string,
 	}
 	return time.Time{}
 }
+
+func appendRawJSONData(jsonFields map[string]interface{}, text []byte) {
+	if jsonFields == nil {
+		jsonFields = common.MapStr{}
+	}
+	jsonFields["message"] = fmt.Sprintf("Raw json data: %s", string(text))
```

Comment:
I think this is going to produce `json.message` in the final event. It looks to me like the `message.Content` field is being populated with the `text` (raw json) when an error occurs.

So I think something upstream in the harvester needs to change to allow the `fields["message"] = message.Content` to be set. I think the issue lies within this code:

https://github.com/elastic/beats/blob/a93acbd9c1d922f4c9e514ec9a3bf6cfb54c0ed2/filebeat/input/log/harvester.go#L272-L307

@ruflin I think we need your Filebeat expertise on this modification.

Reply:
I think what you want is to append a message key in the final event. That should be pretty straightforward to implement.